packages/aztec.js/src/proof/burn/index.js

```diff
@@ -0,0 +1,191 @@
+/**
+ * Constructs AZTEC join-split zero-knowledge proofs
+ *
+ * @module proof.burn
+ */
+
+const BN = require('bn.js');
+const { padLeft } = require('web3-utils');
+
+const verifier = require('./verifier');
+const proofUtils = require('../proofUtils');
+const joinSplit = require('../joinSplit');
+
+const bn128 = require('../../bn128');
+const Keccak = require('../../keccak');
+const {
+    inputCoder,
+    outputCoder,
+} = require('../../abiEncoder');
+
+const { groupReduction } = bn128;
+
+
+const burn = {};
+burn.verifier = verifier;
+
+/**
+ * Generate random blinding scalars, conditional on the AZTEC join-split proof statement
+ *   Separated out into a distinct method so that we can stub this for extractor tests
+ *
+ * @method generateBlindingScalars
+ * @memberof proof.joinSplit
+ * @param {number} n number of notes
+ * @param {number} m number of input notes
+ */
+burn.generateBlindingScalars = (n) => {
+    const m = 1;
+    let runningBk = new BN(0).toRed(groupReduction);
+    const scalars = [...Array(n)].map((v, i) => {
+        let bk = bn128.randomGroupScalar();
+        const ba = bn128.randomGroupScalar();
+        if (i === (n - 1)) {
+            if (n === m) {
```

Comment:
if n === m we should throw an error (a valid burn proof needs at least 1 input note and 1 output note)

Reply:
(probably best to hoist this check to the top of the function, and remove this if condition altogether)

Reply:
On second thoughts, we should probably remove this function in its entireity. 

Burn proofs (and mint proofs) are identical to join split transactions, just with restricted inputs/outputs. Given that they use the same logic, this module should do the following:  

1. Validate that n >= 2  
2. Format the note data so that it is compatible with joinSplit/index.js , and construct the relevant proof data  
3. Format the proof output according to the burn proof's ABI encoding